homeassistant/components/derivative/config_flow.py

```diff
@@ -31,49 +30,55 @@
 )
 
 UNIT_PREFIXES = [
-    {"value": "none", "label": "none"},
-    {"value": "n", "label": "n (nano)"},
-    {"value": "µ", "label": "µ (micro)"},
-    {"value": "m", "label": "m (milli)"},
-    {"value": "k", "label": "k (kilo)"},
-    {"value": "M", "label": "M (mega)"},
-    {"value": "G", "label": "T (tera)"},
-    {"value": "T", "label": "P (peta)"},
+    selector.SelectOptionDict(value="none", label="none"),
+    selector.SelectOptionDict(value="n", label="n (nano)"),
+    selector.SelectOptionDict(value="µ", label="µ (micro)"),
+    selector.SelectOptionDict(value="m", label="m (milli)"),
+    selector.SelectOptionDict(value="k", label="k (kilo)"),
+    selector.SelectOptionDict(value="M", label="M (mega)"),
+    selector.SelectOptionDict(value="T", label="T (tera)"),
+    selector.SelectOptionDict(value="P", label="P (peta)"),
 ]
 TIME_UNITS = [
-    {"value": TIME_SECONDS, "label": "Seconds"},
-    {"value": TIME_MINUTES, "label": "Minutes"},
-    {"value": TIME_HOURS, "label": "Hours"},
-    {"value": TIME_DAYS, "label": "Days"},
+    selector.SelectOptionDict(value=TIME_SECONDS, label="Seconds"),
+    selector.SelectOptionDict(value=TIME_MINUTES, label="Minutes"),
+    selector.SelectOptionDict(value=TIME_HOURS, label="Hours"),
+    selector.SelectOptionDict(value=TIME_DAYS, label="Days"),
 ]
 
 OPTIONS_SCHEMA = vol.Schema(
     {
         vol.Required(CONF_ROUND_DIGITS, default=2): selector.selector(
-            {
-                "number": {
-                    "min": 0,
-                    "max": 6,
-                    "mode": "box",
-                    CONF_UNIT_OF_MEASUREMENT: "decimals",
-                }
-            }
+            selector.SelectorType.NUMBER,
+            selector.NumberSelectorDict(
+                min=0,
+                max=6,
+                mode=selector.NumberSelectorMode.BOX,
+                unit_of_measurement="decimals",
+            ),
+        ),
+        vol.Required(CONF_TIME_WINDOW): selector.selector(
+            selector.SelectorType.DURATION, selector.DurationSelectorDict()
         ),
-        vol.Required(CONF_TIME_WINDOW): selector.selector({"duration": {}}),
         vol.Required(CONF_UNIT_PREFIX, default="none"): selector.selector(
-            {"select": {"options": UNIT_PREFIXES}}
+            selector.SelectorType.SELECT,
+            selector.SelectSelectorDict(options=UNIT_PREFIXES),
         ),
         vol.Required(CONF_UNIT_TIME, default=TIME_HOURS): selector.selector(
-            {"select": {"options": TIME_UNITS}}
+            selector.SelectorType.SELECT,
+            selector.SelectSelectorDict(options=TIME_UNITS),
         ),
     }
 )
 
 CONFIG_SCHEMA = vol.Schema(
     {
-        vol.Required(CONF_NAME): selector.selector({"text": {}}),
+        vol.Required(CONF_NAME): selector.selector(
+            selector.SelectorType.TEXT, selector.TextSelectorDict()
+        ),
```

Comment:
We should be able to shorten these cases?


```suggestion
        vol.Required(CONF_NAME): selector.selector(selector.SelectorType.TEXT),
```

Reply:
oh yeah, thanks for the reminder

Reply:
should we get rid of the empty TypedDicts and just not require an argument at all? I kept them there so that there was consistency across all the selector types to make it easier for the next person to add one

Reply:
Dunno, I think keeping it consistent makes sense, it doesn't harm? So why not 👍